code/game/mecha/mech_fabricator.dm

```diff
@@ -172,34 +239,26 @@
 	atom_say("[I] is complete.")
 	being_built = null
 
-	updateUsrDialog()
+	SSnanoui.update_uis(src)
 	return TRUE
 
-/obj/machinery/mecha_part_fabricator/proc/update_queue_on_page()
-	send_byjax(usr,"mecha_fabricator.browser","queue",list_queue())
-	return
-
-/obj/machinery/mecha_part_fabricator/proc/add_part_set_to_queue(set_name)
-	if(set_name in part_sets)
-		for(var/v in files.known_designs)
-			var/datum/design/D = files.known_designs[v]
-			if(D.build_type & fabricator_type)
-				if(set_name in D.category)
-					add_to_queue(D)
-
-/obj/machinery/mecha_part_fabricator/proc/add_to_queue(D)
+// Add a single part to the queue
+/obj/machinery/mecha_part_fabricator/proc/add_to_queue(datum/design/D)
 	if(!istype(queue))

```

Comment:
Unrelated: I'm not a fan of this many safety checks. There's no way D would be null or a non-design type.
There is absolutely no way `istype(queue)` returns false (like what??).
What are you even going to do with the return values? If it's not TRUE, something went terribly wrong already, and you have bigger problems than to handle  the "sorry couldn't add to queue" case.

I mean, I understand where this is coming from, but it all feels very weird.

Reply:
Yeah I don't think queue would ever not be a list, so this is probably unnecessary. But it was here before so I just left it.